library/std/src/sys/pal/windows/c.rs

```diff
@@ -321,6 +321,21 @@ pub unsafe fn NtWriteFile(
 }
 }
 
+// Use raw-dylib to import ProcessPrng as we can't rely on there being an import library.
+cfg_if::cfg_if! {
+if #[cfg(not(target_vendor = "win7"))] {
+    #[cfg(target_arch = "x86")]
+    #[link(name = "bcryptprimitives", kind = "raw-dylib", import_name_type = "undecorated")]
+    extern "system" {
+        pub fn ProcessPrng(pbdata: *mut u8, cbdata: usize) -> BOOL;
+    }
+    #[cfg(not(target_arch = "x86"))]
+    #[link(name = "bcryptprimitives", kind = "raw-dylib")]
```

Comment:
It looks like using `bcryptprimitives` instead of `bcrypt` broke running on the Wine shipped with Ubuntu 22.04:

```
00f0:err:module:import_dll Library bcryptprimitives.dll (which is needed by L"Z:\\home\\runner\\work\\rustc_codegen_cranelift\\rustc_codegen_cranelift\\build\\example\\arbitrary_self_types_pointers_and_wrappers.exe") not found
  00f0:err:module:LdrInitializeThunk Importing dlls for L"Z:\\home\\runner\\work\\rustc_codegen_cranelift\\rustc_codegen_cranelift\\build\\example\\arbitrary_self_types_pointers_and_wrappers.exe" failed, status c0000135
  "wine" "/home/runner/work/rustc_codegen_cranelift/rustc_codegen_cranelift/./build/example/arbitrary_self_types_pointers_and_wrappers" exited with status ExitStatus(unix_wait_status(13568))
```

Reply:
Looks like Wine only added an implementation for `ProcessPrng` 8 months ago in wine 8.13.

Reply:
What is our support policy for Wine? I'll admit to being loathe to make the Win10 experience worse for the sake of old Wine, especially when it's easy enough to make a `bcryptprimitives.dll` shim.

Reply:
I don't know if we have a support policy for Wine at all. I've worked around this for now by disabling the cross-compilation tests of cg_clif for linux to mingw as those are using wine.

Reply:
For Wine specifically this dll should work, if compiled to somewhere in PATH:

<details><summary>bcryptprimitives.rs</summary>

```rust
#![crate_type = "cdylib"]
#![allow(nonstandard_style)]

#[no_mangle]
pub unsafe extern "system" fn ProcessPrng(mut pbData: *mut u8, mut cbData: usize) -> i32 {
    while cbData > 0 {
        let size = core::cmp::min(cbData, u32::MAX as usize);
        RtlGenRandom(pbData, size as u32);
        cbData -= size;
        pbData = pbData.add(size);
    }
    1
}

#[link(name = "advapi32")]
extern "system" {
    #[link_name = "SystemFunction036"]
    pub fn RtlGenRandom(RandomBuffer: *mut u8, RandomBufferLength: u32) -> u8;
}
```

</details>

Reply:
I don't seem to be able to convince wine to load it. This is what I tried: https://github.com/rust-lang/rustc_codegen_cranelift/commit/cbc0f65deb469b632c766af2f00cd788d3e72ccc

Reply:
Hm, is Wine PATH different from host PATH?

Reply:
Oh, you might need to set `WINEPATH`.

Reply:
Just tried that and it works just fine. Thanks!